passive/SQL injection detection.js

```diff
@@ -0,0 +1,204 @@
+    alertTitle = ["MySQL error Disclosed (script)",
+		  "Postgresql error Disclosed (script)",
+		  "MSSQL error Disclosed (script)",
+		  "Microsoft Access error Disclosed (script)",
+		  "Oracle error Disclosed (script)",
+		  "IBM DB2 error Disclosed (script)",
+		  "Informix error Disclosed (script)",
+		  "Firebird error Disclosed (script)",
+		  "SQLite error Disclosed (script)",
+		  "SAP DB error Disclosed (script)",
+		  "Sybase error Disclosed (script)",
+		  "Ingress error Disclosed (script)",
+		  "Frontbase error Disclosed (script)",
+		  "HSQLDB error Disclosed (script)",
+		  ""]
+    alertDesc = ["A MySQL error was discovered.",
+		 "A Postgresql error was discovered.",
+		 "A MSSQL error was discovered.",
+		 "A Microsoft Access error was discovered.",
+		 "An Oracle error was discovered.",
+		 "An IBM DB2 error was discovered.",
+		 "An Informix error was discovered.",
+		 "A Firebird error was discovered.",
+		 "An SQLite error was discovered",
+		 "A SAP DB error was discovered",
+		 "A Sybase error was discovered",
+		 "An Ingress error was discovered",
+		 "A Frontbase error was discovered",
+		 "A HSQLDB error was discovered",
+		""]
+    alertSolution = ["Ensure proper sanitisation is done on the server side, or don't. I don't care.",
+		    ""]
+    cweId = [0,1]
+    wascId = [0,1]
+
+    mysql = /(SQL syntax.*MySQL|Warning.*mysql_.*|MySqlException \(0x|valid MySQL result|check the manual that corresponds to your (MySQL|MariaDB) server version|MySqlClient\.|com\.mysql\.jdbc\.exceptions)/g
+    postgresql = /(PostgreSQL.*ERROR|Warning.*\Wpg_.*|valid PostgreSQL result|Npgsql\.|PG::SyntaxError:|org\.postgresql\.util\.PSQLException|ERROR:\s\ssyntax error at or near)/g
+    mssql = /(Driver.* SQL[\-\_\ ]*Server|OLE DB.* SQL Server|\bSQL Server.*Driver|Warning.*mssql_.*|\bSQL Server.*[0-9a-fA-F]{8}|[\s\S]Exception.*\WSystem\.Data\.SqlClient\.|[\s\S]Exception.*\WRoadhouse\.Cms\.|Microsoft SQL Native Client.*[0-9a-fA-F]{8})/g
+    msaccess = /(Microsoft Access (\d+ )?Driver|JET Database Engine|Access Database Engine|ODBC Microsoft Access)/g
+    oracle = /(\bORA-\d{5}|Oracle error|Oracle.*Driver|Warning.*\Woci_.*|Warning.*\Wora_.*)/g
+    ibmdb2 = /(CLI Driver.*DB2|DB2 SQL error|\bdb2_\w+\(|SQLSTATE.+SQLCODE)/g
+    informix = /(Exception.*Informix)/g
+    firebird = /(Dynamic SQL Error|Warning.*ibase_.*)/g
+    sqlite = /(SQLite\/JDBCDriver|SQLite.Exception|System.Data.SQLite.SQLiteException|Warning.*sqlite_.*|Warning.*SQLite3::|\[SQLITE_ERROR\])/g
+	sapdb = /(SQL error.*POS([0-9]+).*|Warning.*maxdb.*)/g
+	sybase = /(Warning.*sybase.*|Sybase message|Sybase.*Server message.*|SybSQLException|com\.sybase\.jdbc)/g
+	ingress = /(Warning.*ingres_|Ingres SQLSTATE|Ingres\W.*Driver)/g
+	frontbase = /(Exception (condition )?\d+. Transaction rollback.)/g
+	hsqldb = /(org\.hsqldb\.jdbc|Unexpected end of command in statement \[|Unexpected token.*in statement \[)/g
+
```

Comment:
For your consideration: IMO, these regexs should be dropped into an array:

```JavaScript

dbmsErr = [mysql, postgresq, mysql, etc, etc]

```

Then you can just do one FOR loop to knock through all of these regexp exec calls.

```JavaScript

	for (var i=0; i < dbmsErr.length; i++)
	{
		while (comm = dbmsErr[i].exec(body)) 
		{
			foundSqlErr.push(comm[0]); 
		}

	}

```

Then alert using alertTitle[i] and alertDesc[i] and you chop out all of those Copy/Pasted and nearly identical IF tests.  Food for thought.